stdlib/src/builtin/io.mojo

```diff
@@ -358,28 +358,11 @@ fn _put(x: DType):
 # ===----------------------------------------------------------------------=== #
 
 
-@no_inline
-fn print(
-    *, sep: StringLiteral = " ", end: StringLiteral = "\n", flush: Bool = False
-):
-    """Prints the end value.
-
-    Args:
-        sep: The separator used between elements.
-        end: The String to write after printing the elements.
-        flush: If set to true, then the stream is forcibly flushed.
-    """
-    _put(end)
-    if flush:
-        _flush()
-
-
 @no_inline
 fn print[
-    T: Stringable, *Ts: Stringable
+    *Ts: Stringable
```

Comment:
@Mogball Is this still needed (split first from rest of pack) for compiler optimization/speed, or is this just an artifact of where packs were at the time?

Reply:
This is small valid as a small optimization. Invoking the function with a pack still has overhead if there is a single element. Although I don't really imagine `print` is performance sensitive

Reply:
Let's do the simple, intuitive thing than rather than trying to micro-optimize compile-time perf IMO.

Reply:
It's not compile-time perf, it's generated code perf

Reply:
In the future. Packs should zero-overhead, but it's a TODO atm

Reply:
Knowing this, I believe it's safe to merge this PR as the overhead will disapear at some point without us having to change anything

Reply:
Yep, I just imported it and I'll merge it soon.